packages/mermaid/src/diagrams/class/classTypes.ts

```diff
@@ -14,6 +16,95 @@ export interface ClassNode {
   tooltip?: string;
 }
 
+export class ClassMember {
+  id!: string;
+  cssStyle!: string;
+  memberType!: string;
+  visibility!: string;
+  classifier!: string;
+  parameters!: string;
+  returnType!: string;
+
+  constructor(input: string, memberType: string) {
+    this.memberType = memberType;
+    this.visibility = '';
+    this.classifier = '';
+    this.parseMember(input);
+  }
+
+  getDisplayDetails() {
+    let displayText = this.visibility + parseGenericTypes(this.id);
+    if (this.memberType === 'method') {
+      displayText += '(' + parseGenericTypes(this.parameters.trim()) + ')';
+      if (this.returnType) {
+        displayText += ' : ' + parseGenericTypes(this.returnType);
+      }
+    }
+
+    displayText = displayText.trim();
+    const cssStyle = this.parseClassifier();
+
+    return {
+      displayText,
+      cssStyle,
+    };
+  }
+
+  parseMember(input: string) {
```

Comment:
If we only ever use this function internally, we might want to make it [`private`](https://www.typescriptlang.org/docs/handbook/2/classes.html#private).

Reply:
I'm marking this as resolved, since it's not a big deal if we don't add `private`.

But you're welcome to add it if you want:

```suggestion
  private parseMember(input: string) {
```

(`private` is a TypeScript only thing, it won't affect the generated JavaScript. Using [`#parseMember()` is how to make private methods in JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields), but it's only supported by newer versions of JavaScript, so it's probably not worth using in Mermaid)